K2Bridge/Utils/DateMathParser.cs

```diff
@@ -0,0 +1,90 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+// See LICENSE file in the project root for full license information.
+
+namespace K2Bridge.Visitors
+{
+    using System;
+    using System.Text.RegularExpressions;
+
+    /// <content>
+    /// Parser for the Date Math expressions.
+    /// </content>
+    public class DateMathParser
+    {
+        public static string ParseDateMath(string expr)
+        {
+            // See Wiki or ES docs for details on date math syntax
+            var rgx = new Regex(
+                @"^
+                        (?: (?<anchor>now) | (?: (?<anchor>.+?) (?:\|\||$)) )
+                        (?: (?<rangeSign>[+-]) (?<rangeValue>\d+) (?<rangeUnit>[a-zA-Z]))*
+                        (?: (?:\/ (?<rounding>y|M|w|d|h|H|m|s)))?", RegexOptions.IgnorePatternWhitespace);
+            var match = rgx.Match(expr);
+
+            var kexpr = string.Empty;
+
+            if (!match.Success)
+            {
+                return kexpr;
+            }
+
+            // The date literal, ie.g. YYYY-MM-DD or "now"
+            var anchor = match.Groups["anchor"].Value;
+            if (anchor.Equals("now"))
+            {
+                kexpr = "now()";
+            }
+            else
+            {
+                kexpr = $"make_datetime('{anchor}')";
+            }
+
+            // Date operation, e.g. -1M, +3d, etc.
+            if (match.Groups["rangeSign"].Success)
```

Comment:
You don't need this if anymore, if there aren't any range expressions, the Captures.Count would be 0 and you won't get in the loop

Reply:
Right, fixed